c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -404,7 +404,7 @@ class DeviceCachingAllocator {
   // See free() for this thing's purpose
   std::vector<Block*> needs_events_deferred_until_no_capture;
   // outstanding cuda events
-  std::deque<std::pair<cudaEvent_t, Block*>> cuda_events;
+  std::unordered_map<cuda::CUDAStream, std::deque<std::pair<cudaEvent_t, Block*>>> cuda_events;
```

Comment:
use `ska::flat_hash_map` here, it's faster?

Reply:
Will change! I deferred to the fact that this code was already using STL maps elsewhere, but I'm always happy to use a properly-optimized hash map instead of the STL.

Reply:
Should I change the other maps in this file for consistency while I'm here, or leave that for a followup PR?